signature/cose/conformance_test.go

```diff
@@ -0,0 +1,216 @@
+package cose
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/json"
+	"os"
+	"reflect"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-core-go/testhelper"
+	"github.com/veraison/go-cose"
+)
+
+type sign1 struct {
+	SigningTime        int64       `json:"signingTime"`
+	Expiry             int64       `json:"expiry"`
+	Payload            string      `json:"payload"`
+	ProtectedHeaders   *cborStruct `json:"protectedHeaders"`
+	UnprotectedHeaders *cborStruct `json:"unprotectedHeaders"`
+	Output             cborStruct  `json:"expectedOutput"`
+}
+
+type cborStruct struct {
+	CBORHex  string `json:"cborHex"`
+	CBORDiag string `json:"cborDiag"`
+}
+
+func TestConformance(t *testing.T) {
+	data, err := os.ReadFile("testdata/conformance.json")
+	if err != nil {
+		t.Fatalf("os.ReadFile() failed. Error = %s", err)
+	}
+	var sign1 sign1
+	err = json.Unmarshal(data, &sign1)
+	if err != nil {
+		t.Fatalf("json.Unmarshal() failed. Error = %s", err)
+	}
+	testSign(t, &sign1)
+	testVerify(t, &sign1)
+}
+
+// testSign does conformance check on COSE_Sign1_Tagged
+func testSign(t *testing.T, sign1 *sign1) {
+	signRequest, err := getSignReq(sign1)
+	if err != nil {
+		t.Fatalf("getSignReq() failed. Error = %s", err)
+	}
+	env := envelope{}
+	encoded, err := env.Sign(signRequest)
+	if err != nil || len(encoded) == 0 {
+		t.Fatalf("Sign() faild. Error = %s", err)
+	}
+	newMsg := generateSign1(env.base)
+	got, err := newMsg.MarshalCBOR()
+	if err != nil {
+		t.Fatalf("MarshalCBOR() faild. Error = %s", err)
+	}
+
+	// sign1.Output.CBORHex is a manually computed CBOR hex used as ground
+	// truth in the conformance test.
+	want := hexToBytes(sign1.Output.CBORHex)
+	if !bytes.Equal(want, got) {
+		t.Fatalf("unexpected output:\nwant: %x\n got: %x", want, got)
+	}
+
+	// Verify using the same envelope struct
+	_, _, err = env.Verify()
```

Comment:
Do we need to create a new envelope by calling ParseEnvelope() before Verify()?

Reply:
Oh yeah, let me update the comment, the part you mentioned is covered in the testVerify() function